File: cpp/766-toeplitz_matrix.cpp
#define min2(a, b) (a < b ? a : b)
class Solution {
public:
    bool isToeplitzMatrix(vector<vector<int>>& matrix) {
        if(matrix.size() == 0 || matrix[0].size() == 0) return false;
        int m = matrix.size(), n = matrix[0].size(), v = matrix[0][0];
        for(int i = 1; i < min2(m, n); ++i) if(matrix[i][i] != v) return false;
        return true;
    }
};

File: python/766-toeplitz_matrix.py
class Solution:
    def isToeplitzMatrix(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: bool
        """
        if len(matrix) == 0 or len(matrix[0]) == 0:
            return False
        v = matrix[0][0]
        for i in range(1, min(len(matrix), len(matrix[0]))):
            if v != matrix[i][i]:
                return False
        return True